backend/app/Http/Controllers/UserController.php

```diff
@@ -5,36 +5,19 @@
 use App\Models\User;
 use Illuminate\Http\Request;
 use Illuminate\Support\Str;
+use Illuminate\Support\Facades\Auth;
 
 class UserController extends Controller
 {
-    /**
-     * Display a listing of the resource.
-     *
-     * @return \Illuminate\Http\Response
-     */
-    public function index()
+    public function index(Request $request)
     {
-        return User::whereNotIn('Role', [0])->get();
+        return $request->user()->tokenCan('adminAccess') ?
+            User::whereNotIn('Role', [0])->get() :
```

Comment:
make json/array of roles, change accordingly
```suggestion
            $ROLE = [
                "SUPER_ADMIN" => 0,
                "ADMIN" => 1,
                "LOWER_ADMIN" => 2,
            ];
            
            User::whereNotIn('Role', [$ROLE['LOWER_ADMIN']])->get() :
```